bluemira/base/builder.py

```diff
@@ -111,6 +113,16 @@ def __init__(
         self.name = self.build_config.get(
             "name", _remove_suffix(self.__class__.__name__, "Builder")
         )
+        self._build = self.build
+        self.build = self._build_wrapper
+
+    def _build_wrapper(self) -> Component:
```

Comment:
I guess decorators have to be imported and applied to subclasses, is that why this is like this?

Reply:
exactly yeah, I couldn't think of a way to do this implicitly without giving the reactor designer something else to do.